utils/cache/filecache/filecache.go

```diff
@@ -0,0 +1,210 @@
+package filecache
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+type Cache struct {
+	entries       iEntryMap
+	entriesLM     iLockMap
+	entryProvider IEntryRetriever
+	fs            filesystem.FS
+	cfg           *FileCacheConfig
+	cancelStore   *parallelisation.CancelFunctionStore
+	closed        atomic.Bool
+}
+
+func (c *Cache) gc(ctx context.Context, _ time.Time) {
+	if err := parallelisation.DetermineContextError(ctx); err != nil {
+		return
+	}
+
+	c.entries.Range(func(key string, entry ICacheEntry) {
+		if entry.IsExpired() && c.entriesLM.TryLock(key) {
+			if err := entry.Delete(ctx); err == nil {
+				c.entries.Delete(key)
+				defer func() {
+					c.entriesLM.Unlock(key)
+					c.entriesLM.Delete(key)
+				}()
+			}
+		}
+	})
+}
+
+func (c *Cache) checkIfClosed() error {
+	if c.closed.Load() {
+		return commonerrors.New(commonerrors.ErrConflict, "cache is closed")
+	}
+
+	return nil
+}
+
+func (c *Cache) Has(ctx context.Context, key string) (bool, error) {
+	if err := c.checkIfClosed(); err != nil {
+		return false, err
+	}
+
+	return c.entries.Exists(key), nil
+}
+
+func (c *Cache) Store(ctx context.Context, key string) error {
+	return c.StoreWithTTL(ctx, key, c.cfg.TTL)
+}
+
+func (c *Cache) StoreWithTTL(ctx context.Context, key string, ttl time.Duration) error {
+	if err := c.checkIfClosed(); err != nil {
+		return err
+	}
+
+	if ttl == 0 {
+		return commonerrors.New(commonerrors.ErrInvalid, "TTL for an entry can't be zero")
+	}
+
+	if c.entries.Exists(key) {
+		return commonerrors.Newf(commonerrors.ErrExists, "cache entry %s already exists", key)
```

Comment:
```suggestion
		return commonerrors.Newf(commonerrors.ErrExists, "cache entry '%s' already exists", key)
```

Reply:
This applied everywhere in this file.

It is a good practice to wrap them in quotes as if they are unset then you can tell by the error because it will say something like `entry '' already set` so you can tell the issue is that it is empty.

You can even do `%q` which will automatically add the quotes. Adrien likes to use `%v` for all of them just to keep it simple though.